### ISSUE TITLE
[vcpkg upgrade] Running `vcpkg upgrade` on Windows will also trigger upgrades for linux triplets

### ISSUE THREAD

**Issue by alexkaratarakis** [microsoft/vcpkg#3453]
If you have a vcpkg instance that has both windows and linux triplets, then `vcpkg upgrade` will try to upgrade all the triplets.

So, if you run `vcpkg upgrade` from windows, it will uninstall ports installed on the linux triplet and then fail to reinstall them.

**Comment by MikeGitb** [comment]
This is extremely annoying and makes shareing a vcpkg instance between WSL and regular windows almost useless. 

I think the most straight forward fix would be to add a triplet option to vcpkg upgrade, such that only packages from aparticular triplet would be updated. 

Another, more general solution would be to allow regular/globbing expressions when specifying the packages to update. E.g.  `vcpkg upgrade .*:x64-linux`

**Comment by JackBoosY** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen this issue.

**Comment by MikeGitb** [comment]
Yes, this is still a problem. Please re-open and address or mark as "Won't Fix".

**Comment by tekert** [comment]
This is still a problem, i just wiped >100 libs trying to upgrade, this should be a simple fix, just check for tripplets  and dont upgrade/delete them from another os.

**Comment by jimwang118** [comment]
This issue has been fixed in latest version of vcpkg. 
![Image](https://github.com/user-attachments/assets/b69a8f6f-3b99-4d64-a21b-72d1417ed7b8)

**Comment by dg0yt** [comment]
> This issue has been fixed in latest version of vcpkg.

This has not been fixed. I guess it won't ever be fixed. It is simply unsupported. Basically, it doesn't make sense to mix different host triplets in the same installed tree.

**Comment by dg0yt** [comment]
This should be wont-fix.
